### PR TITLE
Semantic Search API Gated Public Preview

### PR DESCRIPTION
**Pull request by Yahnoosh**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  Already deployed.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  We're flexible.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 1]
```diff
@@ -245,6 +251,93 @@
...
+            "name": "speller",
+            "in": "query",
+            "type": "string",
+            "enum": [
+              "none",
+              "lexicon"
+            ],
+            "x-ms-enum": {
+              "name": "Speller",
+              "modelAsString": false,
```
Comment (jhendrixMSFT) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:293]: To confirm, there's no anticipation of future values for this enum?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 2]
```diff
@@ -1230,10 +1404,87 @@
...
+    },
+    "Speller": {
+      "type": "string",
+      "enum": [
+        "none",
+        "lexicon"
+      ],
+      "x-ms-enum": {
+        "name": "Speller",
+        "modelAsString": false,
```
Comment (jhendrixMSFT) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1425]: To confirm, there's no anticipation of future values for this enum?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 3]
```diff
@@ -1230,10 +1404,87 @@
...
+    },   
+    "QueryLanguage": {
+      "type": "string",
+      "enum": [
+        "none",
+        "en-us"        
+      ],
+      "x-ms-enum": {
+        "name": "QueryLanguage",
+        "modelAsString": false,
```
Comment (jhendrixMSFT) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1473]: To confirm, there's no anticipation of future values for this enum?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 4]
```diff
@@ -992,6 +1085,54 @@
...
+          "type": "string",
+          "readOnly": true,
+          "description": "The text passage extracted from the document contents as the answer.",
+          "x-ms-client-name": "Text"
+        },
+        "highlights": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Same text passage as in the Text property with highlighted text phrases most relevant to the query.",
+          "x-ms-client-name": "Text"
```
Comment (jhendrixMSFT) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1112]: This collides with the value on line 1106.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 5]
```diff
@@ -992,6 +1085,54 @@
...
+          "type": "string",
+          "readOnly": true,
+          "description": "A representative text passage extracted from the document most relevant to the search query.",
+          "x-ms-client-name": "Text"
+        },
+        "highlights": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Same text passage as in the Text property with highlighted phrases most relevant to the query.",
+          "x-ms-client-name": "Text"
```
Comment (jhendrixMSFT) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1130]: This collides with the value on line 1124.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
There are some name collisions that need to be resolve.  Please also look at the CI failures and fix the relevant ones.

Has this change been reviewed by the API review board?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 3 above)
Comment (Yahnoosh) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1473]: Is the recommendation to use modelAsString for all enums that might have more values in the future?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 5 above)
Comment (Yahnoosh) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1130]: sorry, sloppy copy-paste

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 3 above)
Comment (jhendrixMSFT) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json]: Yes, if you anticipate additions in the future then using `modelAsString:true` is more versioning friendly.

**Comment by jhendrixMSFT** [comment]
Can you please address the prettier and spell check failures?

**Pull request closed without merging**